src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// Specifies the property name that is present in the JSON when serializing and deserializing.
+    /// This overrides any naming policy specified by <see cref="JsonNamingPolicy"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public sealed class JsonPropertyNameAttribute : JsonAttribute
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="JsonPropertyNameAttribute"/> with the specified property name.
+        /// </summary>
+        /// <param name="propertyName">The name of the property.</param>
+        public JsonPropertyNameAttribute(string propertyName)
+        {
+            Name = propertyName;
```

Comment:
Is there any way to detect/flag that the property name passed in here is identical to the actual property name and hence his is unnecessary/redundant? Presumably, using this attribute leads to slower serialization than using the default property name.

Reply:
It is not redundant - if specified then the value will be used even with a naming policy.

Reply:
Good point.